server/utils/SWUStats/SWUStatsHandler.ts

```
@@ -0,0 +1,239 @@
+import type Game from '../../game/core/Game';
+import type { Player } from '../../game/core/Player';
+import type { ISetId } from '../../game/Interfaces';
+import type { IDecklistInternal } from '../deck/DeckInterfaces';
+
+interface SWUstatsGameResult {
+    apiKey: string;
+    winner: number; // 1 or 2
+    firstPlayer: number; // 1 or 2
+    round: number;
+    winHero: string;
+    loseHero: string;
+    winnerDeck?: string;
+    loserDeck?: string;
+    winnerHealth: number;
+    player1: string; // JSON string with leader/base info
+    player2: string; // JSON string with leader/base info
+    p1DeckLink: string;
+    p2DeckLink: string;
+    p1id?: string;
+    p2id?: string;
+    gameName: string;
+    sequenceNumber?: number;
+}
+
+interface PlayerData {
+    gameId?: string;            // Unique game identifier (optional)
+    gameName?: string;          // Custom game name (optional)
+    deckId: string;             // The last part of the deck link
+    leader: string;             // Leader id (FFG UID format)
+    base: string;               // Base id (FFG UID format)
+    turns: number;              // Number of turns played
+    result: number;             // 1 if this player won, 0 for loss
+    firstPlayer: number;        // 1 if this player went first, 0 otherwise
+    opposingHero: string;       // Opponent's leader id (FFG UID format)
+    opposingBaseColor: string;  // Opponent's base color (Red, Blue, Yellow, Green, Colorless)
+    deckbuilderID?: string;     // Deckbuilder user ID
+}
+
+export class SWUstatsHandler {
+    private readonly apiUrl: string;
+    private readonly apiKey: string;
+
+    public constructor() {
+        // Use environment variable for API URL, defaulting to the known endpoint
+        this.apiUrl = process.env.SWUStatsURL;
+        this.apiKey = process.env.SWUStatsAPIKey;
+        if (!this.apiKey) {
+            logger.warn('SWUstatsHandler: No API key configured. Stats may not be accepted by SWUstats.');
```

This should probably throw an error and block the server from starting if we're not in local dev mode